### PR TITLE
REU: Update Port Terminology

### PR DESCRIPTION
**Pull request by ebrickey**
This branch is a culmination of all the REU's work to update the port terminology as per Jira: https://gem5.atlassian.net/browse/GEM5-9

### CODE REVIEW

File: src/dev/arm/smmu_v3.cc [hunk 1]
```diff
@@ -311,7 +311,7 @@ SMMUv3::runProcessTiming(SMMUProcess *proc, PacketPtr pkt)
                     action.pkt->getAddr(), action.pkt->getSize());
 
             if (packetsToRetry.empty() &&
-            masterPort.sendTimingReq(action.pkt)) {
+                    masterPort.sendTimingReq(action.pkt)) {
```
Comment (powerjg) [inline src/dev/arm/smmu_v3.cc:314]: This needs to be reverted :)

File: src/arch/arm/fastmodel/FastModel.py [hunk 2]
```diff
@@ -75,21 +75,21 @@ def __init__(self, width, desc):
         super(VectorAmbaInitiatorSocket, self).__init__(
                 my_role, desc, is_source=True)
 
-class ScMasterPort(Port):
```
Comment (powerjg) [inline src/arch/arm/fastmodel/FastModel.py:78]: Let's back out this change. This is changing something quite different than everything else in this commit.

**Review by powerjg — COMMENTED** [review]
A few changes before we push :). It's better to do it now then wait for someone else to say the same things....

In  https://github.com/darchr/gem5/pull/187/commits/5bcdabf5628b31c4b71ca6448f6ac2372a9dad6b
- the Learning gem5 changes should be in the Learning gem5 commit. It should be in https://github.com/darchr/gem5/pull/187/commits/9c3d2c3be132cdd343db969e35cae5df4484dd77
- It should be titled "mem,sim: ..."

For https://github.com/darchr/gem5/pull/187/commits/ef3333eb2e297624166850e0e5f11c499c7f28a3

Let's update the commit message to
"mem: deprecate SlavePort and MasterPort classes

After this change, if you use these classes or inherit from these classes, the compiler will now give you a warning that these names are deprecated. Instead, you should use ResponsePort and RequestPort, respectively.

This patch simply deprecates these names. The following patches will convert all of the code in gem5 to use these new names. The first step is converting the class names and the uses of these classes, then we will update the variable names to be more precise as well."

File: src/arch/arm/fastmodel/FastModel.py (hunk 2 above)
Comment (shparekh00) [reply src/arch/arm/fastmodel/FastModel.py:78]: So should I keep it as a separate commit?
Or undo the changes?

Comment (powerjg) [reply src/arch/arm/fastmodel/FastModel.py]: Sorry for the slow reply here :). Probably back out... this is one that we might have to do something special for

**Pull request closed without merging**